src/main/scala/chisel3/util/Reg.scala

```diff
@@ -42,7 +42,7 @@ object ShiftRegister
     * val regDelayTwo = ShiftRegister(nextVal, 2, ena)
     * }}}
     */
-  def apply[T <: Data](in: T, n: Int, en: Bool = true.B): T = ShiftRegisters(in, n, en).last
+  def apply[T <: Data](in: T, n: Int, en: Bool = true.B): T = ShiftRegisters(in, n, en).lastOption.getOrElse(in)
```

Comment:
`n = 0` -> `ShiftRegisters` returns an empty `Seq`, `ShiftRegister` will directly return the input wire.